pkg/compiler/inline.go

```diff
@@ -129,52 +144,87 @@ func (c *codegen) inlineCall(f *funcScope, n *ast.CallExpr) {
 	c.pkgInfoInline = c.pkgInfoInline[:len(c.pkgInfoInline)-1]
 }
 
-func (c *codegen) processStdlibCall(f *funcScope, args []ast.Expr, hasEllipsis bool) {
+func (c *codegen) processStdlibCall(f *funcScope, args []ast.Expr, hasEllipsis bool) []*stackitem.Type {
 	if f == nil {
-		return
+		return nil
 	}
 
+	var eventParams []*stackitem.Type
 	if f.pkg.Path() == interopPrefix+"/runtime" && (f.name == "Notify" || f.name == "Log") {
-		c.processNotify(f, args, hasEllipsis)
+		eventParams = c.processNotify(f, args, hasEllipsis)
 	}
 
 	if f.pkg.Path() == interopPrefix+"/contract" && f.name == "Call" {
 		c.processContractCall(f, args)
 	}
+	return eventParams
 }
 
-func (c *codegen) processNotify(f *funcScope, args []ast.Expr, hasEllipsis bool) {
+// processNotify checks whether notification emitting rules are met and returns expected
+// notification signature if found.
+func (c *codegen) processNotify(f *funcScope, args []ast.Expr, hasEllipsis bool) []*stackitem.Type {
 	if c.scope != nil && c.isVerifyFunc(c.scope.decl) &&
 		c.scope.pkg == c.mainPkg.Types && (c.buildInfo.options == nil || !c.buildInfo.options.NoEventsCheck) {
 		c.prog.Err = fmt.Errorf("runtime.%s is not allowed in `Verify`", f.name)
-		return
+		return nil
 	}
 
 	if f.name == "Log" {
-		return
+		return nil
 	}
 
 	// Sometimes event name is stored in a var. Or sometimes event args are provided
 	// via ellipses (`slice...`).
-	// Skip in this case.
+	// Skip in this case.  Also, don't enforce runtime.Notify parameters conversion.
 	tv := c.typeAndValueOf(args[0])
 	if tv.Value == nil || hasEllipsis {
-		return
+		return nil
 	}
 
 	params := make([]string, 0, len(args[1:]))
+	vParams := make([]*stackitem.Type, 0, len(args[1:]))
 	for _, p := range args[1:] {
-		st, _, _ := c.scAndVMTypeFromExpr(p)
+		st, vt, _ := c.scAndVMTypeFromExpr(p)
 		params = append(params, st.String())
+		vParams = append(vParams, &vt)
 	}
 
 	name := constant.StringVal(tv.Value)
 	if len(name) > runtime.MaxEventNameLen {
 		c.prog.Err = fmt.Errorf("event name '%s' should be less than %d",
 			name, runtime.MaxEventNameLen)
-		return
+		return nil
+	}
+	var eventFound bool
+	for _, e := range c.buildInfo.options.ContractEvents {
+		if e.Name == name && len(e.Parameters) == len(vParams) {
+			eventFound = true
+			for i, scParam := range e.Parameters {
+				expectedType := scParam.Type.ToStackitemType(true) // runtime.Notify creates immutable deep copy anyway, so use the same constraint here to check the resulting notification list if --no-events option is off.
+				// No need to cast if the desired type is unknown.
+				if expectedType == stackitem.AnyT || expectedType == stackitem.InteropT ||
+					// No need to cast if actual parameter type matches the desired one.
+					*vParams[i] == expectedType ||
+					// expectedType doesn't contain Buffer anyway, but if actual variable type is ByteArray,
+					// then runtime.Notify will convert it to Buffer automatically, thus no need to emit conversion code.
```

Comment:
It's the other way around.

Reply:
Yes, the comment is fixed. The code was written correctly.